mcs/class/System/Test/System.Diagnostics/PerformanceCounterTest.cs

```diff
@@ -0,0 +1,363 @@
+			CounterCreationData averageCount64 = new CounterCreationData();
+			averageCount64.CounterType = PerformanceCounterType.AverageCount64;
+			averageCount64.CounterName = name;
+			counterDataCollection.Add(averageCount64);
+
+			// Add the base counter.
+			CounterCreationData averageCount64Base = new CounterCreationData();
+			averageCount64Base.CounterType = PerformanceCounterType.AverageBase;
+			averageCount64Base.CounterName = name + "Base";
+			counterDataCollection.Add(averageCount64Base);
+
+			// Create the category.
+			PerformanceCounterCategory.Create(name + "Category",
+			"Demonstrates usage of the AverageCounter64 performance counter type.",
+			PerformanceCounterCategoryType.SingleInstance, counterDataCollection);
+
+			Assert.True(PerformanceCounterCategory.Exists(name + "Category"));
+
+			if (verbose)
+				Console.WriteLine("PerformanceCounterCategory_CreateCategory end");
+		}
+
+		[Test]
+		public void PerformanceCounter_CreateCounter_Count0()
+		{
+			if (verbose)
+				Console.WriteLine("PerformanceCounter_CreateCounter_Count0 start");
+			Assert.AreEqual(0, a.counterSample.RawValue);
+			a.counterSample.Increment();
+			Assert.AreEqual(1, a.counterSample.RawValue);
+			if (verbose)
+				Console.WriteLine("PerformanceCounter_CreateCounter_Count0 end");
+		}
+
+		public bool verbose = Environment.GetEnvironmentVariable("V") != null;
+
+		[Test]
+		public void PerformanceCounter_InstanceNames()
+		{
+			if (verbose)
+				Console.WriteLine("PerformanceCounter_InstanceNames start");
+			var names = a.pcc.GetInstanceNames();
+			if (verbose)
+				Console.WriteLine($"\nPerformanceCounter_InstanceNames {a.name} {a.pcc} {names} {names.Length}");
+			Assert.That (new [] { a.name }, Is.EquivalentTo (names));
+			if (verbose)
+				Console.WriteLine("PerformanceCounter_InstanceNames end");
+		}
+
+		[Test]
+		public void PerformanceCounter_Counters()
+		{
+			if (verbose)
+				Console.WriteLine("PerformanceCounter_Counters start");
+			var a = new Common();
+			a.DoCommon();
+			var counters = a.pcc.GetCounters(a.name);
+			int i = 0;
+			if (verbose)
+				Console.WriteLine($"\nPerformanceCounter_Counters {a.name} {counters} {counters.Length}");
+			Assert.AreEqual(2, counters.Length);
+			foreach (var b in counters)
+			{
+				var category = b.CategoryName;
+				var name = b.CounterName;
+				var instance = b.InstanceName;
+				var value = b.RawValue;
+				if (verbose)
+					Console.WriteLine($"i:{i} category:{category} counter:{name} instance:{instance} value:{value}");
+				++i;
+				Assert.True(name == a.name || name == a.baseName);
+				Assert.AreEqual(a.name, instance);
+				Assert.AreEqual(a.category, category);
+				Assert.AreEqual(0, value);
+			}
+			if (verbose)
+				Console.WriteLine("PerformanceCounter_Counters end");
+		}
+	}
+
+// from MSDN
+// https://msdn.microsoft.com/en-us/library/system.diagnostics.countercreationdata(v=vs.110).aspx
+
+	[TestFixture]
+	public class MSDN_PerformanceCounterTest
```

Comment:
I'd merge the tests from this class into `PerformanceCounterTests`, there's no real reason to make a separate class.